File: Algorithms/150_EvaluateReversePolishNotion.java
public class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stk = new Stack<Integer>();
        for (int i=0; i < tokens.length; ++i) {
            if (tokens[i].equals("+")) {
                int a = stk.peek();
                stk.pop();
                int b = stk.peek();
                stk.pop();
                stk.push(b + a);
            } else if (tokens[i].equals("-")) {
                int a = stk.peek();
                stk.pop();
                int b = stk.peek();
                stk.pop();
                stk.push(b - a);
            } else if (tokens[i].equals("*")) {
                int a = stk.peek();
                stk.pop();
                int b = stk.peek();
                stk.pop();
                stk.push(b * a);
            } else if (tokens[i].equals("/")) {
                int a = stk.peek();
                stk.pop();
                int b = stk.peek();
                stk.pop();
                stk.push(b / a);
            } else {
                stk.push(Integer.parseInt(tokens[i]));
            }
        }
        return stk.peek();
    }
}

File: Algorithms/py/150.EvaluateReversePolishNotion.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for c in tokens:
            if c in ['+', '-', '*', '/']:
                a = stack.pop()
                b = stack.pop()
                if c == '+':
                    stack.append(b + a)
                elif c == '-':
                    stack.append(b - a)
                elif c == '*':
                    stack.append(b * a)
                elif c == '/':
                    stack.append((abs(b) // abs(a)) * (1 if a * b > 0 else -1))
            else:
                stack.append(int(c))
        return stack.pop()